bosh/plugin.go

```diff
@@ -4,27 +4,64 @@ import (
 	"encoding/json"
 	"log"
 	"net/http"
+
+	"github.com/cloudfoundry/bosh-agent/jobsupervisor/monit"
 )
 
 type plugin struct {
-	report report
+	report          report
+	monitHTTPClient monit.Client
 }
 
-func NewPlugin(hostname, jobSpecPath string) *plugin {
+func NewPlugin(hostname, jobSpecPath, platform string) (*plugin, error) {
 	report := newReport(hostname)
 
 	if jobSpec, err := loadSpec(jobSpecPath); err != nil {
 		log.Printf("error loading job spec from %q: %v\n", jobSpecPath, err)
+		return nil, err
 	} else {
-		report.AddJob(jobSpec)
+		report.SetInstanceSpecData(jobSpec)
+	}
+
+	var monitHTTPClient monit.Client
+	var err error
+
+	if platform == "ubuntu" {
+		monitHTTPClient, err = newMonitHTTPClient()
+		if err != nil {
+			log.Printf("error creating monit HTTP client: %v\n", err)
+			return nil, err
+		}
 	}
 
-	return &plugin{report}
+	return &plugin{report, monitHTTPClient}, nil
 }
 
 func (p *plugin) Report(w http.ResponseWriter, r *http.Request) {
-	if err := json.NewEncoder(w).Encode(p.report); err != nil {
+	err := p.updateReport()
+
+	if err != nil {
+		log.Printf("error updating report: %v\n", err)
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+	}
+
+	if err = json.NewEncoder(w).Encode(p.report); err != nil {
 		log.Printf("error encoding report: %v\n", err)
 		http.Error(w, err.Error(), http.StatusInternalServerError)
 	}
 }
+
+func (p *plugin) updateReport() error {
+	if p.monitHTTPClient == nil {
+		return nil
+	}
+
+	status, err := p.monitHTTPClient.Status()
```

Comment:
While this seems to be working fine according to your screenshots, it might actually be problematic to get the status synchronously from within this HTTP handler. The reason I'm saying this is that the Scope Probe expects any request to a plugin to respond within 500ms otherwise it marks the plugin as failing inside the App [1]. For that reason it might be better to start a Go routine during startup and have that asynchronously refresh the status inside a package-level variable. The HTTP handler could then reference that variable.

That asynchronous behavior could fit in nicely into the `bosh/monit` package I suggested above, i.e. our client would start the Go routine internally and callers would just use a method like `func (c client) Services(group string) ([]string, error)` that always returns the most recently fetched list of services. One more advantage of this would be that none of the `bosh-agent` packages would "leak" out of the client implementation.

[1] https://github.com/weaveworks/scope/tree/master/examples/plugins#registering-plugins